### PR TITLE
Homework 2 Version 1 Submission

### PR DESCRIPTION
**Pull request by hotz1**
Here is our group's submission for Version 1 of HW2.

### CODE REVIEW

File: report/main.qmd [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+{{< pagebreak >}}
+
+```{=tex}
+\pagenumbering{arabic}
+```
+
+# Introduction
+
+Strong evidence has linked habitual intake of sugar-sweetened beverages (SSBs) with weight gain and a higher risk of type 2 diabetes mellitus, cardiovascular diseases, and even cancer @Malik2022. Multiple policy interventions to inhibit consumption of SSBs have been proposed @Krieger2021, including taxation of SSBs, limiting access to SSBs in schools and healthcare facilities, and adding informative package labelling. As policymakers create potential interventions to limit SSB consumption, there is a need for data-driven methods of determining their efficacy.
```
Comment (sarahmasri) [inline report/main.qmd:33]: This whole paragraph is great

File: report/main.qmd [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+This study aims to evaluate five interventions which intend to shift consumer behaviour towards zero-calorie beverages. These interventions include three information-based interventions and two financial interventions.
```
Comment (sarahmasri) [inline report/main.qmd:35]: I think the primary objective is to compare two interventions (labelling calorie content) and then price discounts are a secondary research question

File: report/main.qmd [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+The main statistical question is whether these interventions are associated with differences in the average proportion of daily sales of zero-calorie beverages. Secondary questions include whether there are significant interactions between interventions and site effects, and whether a combination of interventions was more effective than the individual interventions.
+
+# Data Description and Summaries
+
+The data come from an interrupted quasi-experimental study on sales of sweetened and zero-calorie beverages which was conducted in three United States hospitals over a 30-week period. Each observation represents the total daily sales of bottled beverages across multiple storefronts in a single site. Sales were recorded automatically by point of sale terminals in Site A, and manually by cashiers at Sites B and C. Sales were recorded every day from the start of the experiment until the final day, including weekends and holidays.
```
Comment (sarahmasri) [inline report/main.qmd:41]: The data come from an interrupted time-series multi-site quasi-experimental

File: report/main.qmd [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+The following variables were collected: the day of the experiment (e.g. Day 1), the corresponding weekday, the site, and the ongoing intervention. The day counter is a positive integer. Categorical variables include weekday with seven levels, site with three levels, and intervention with nine levels. The daily sales metrics include the number of bottled zero-calorie and sweetened beverages sold per site. These are each non-negative integer variables. The data also includes the total number of beverages sold per site, which is another non-negative integer value.
```
Comment (sarahmasri) [inline report/main.qmd:43]: I don’t think the example (e.g. Day 1) is needed say day of experiment is a fairly clear metric- maybe “day of measurement” if you wan’t to be more specific?

Also maybe use variable/factor/measurement instead of "counter"

File: report/main.qmd [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+Some observations additionally include the sales of 100% juice, orange juice, and sports drinks, which are recorded as integers. However, these counts are only consistently recorded at Site A, and rarely recorded for Site B and C. There are nine days which are missing counts for the bottled zero-calorie and sugared beverages. Each of these nine observations contains missing data for every sales-related column. Of these nine observations, seven occur in the final week of the study at Site B, and two occur mid-study at Site C.
```
Comment (sarahmasri) [inline report/main.qmd:45]: This sentence "There are nine days which are missing counts for the bottled zero-calorie and sugared beverages" is a little confusing. Maybe rephrase?

File: report/main.qmd [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+# Exploratory Analysis
```
Comment (sarahmasri) [inline report/main.qmd:47]: It might be helpful to show examples of each of the recommended figures- if not in the main report then perhaps in the appendix

File: report/main.qmd [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+
+Exploratory data analysis can visualize relationships among variables, and informally give some insight on what results can be expected. Based on the data and statistical questions, we recommend the following figures:
```
Comment (sarahmasri) [inline report/main.qmd:49]: “An exploratory data analysis can…”

Add that the EDA can check distributional properties

If you want to avoid saying 'we' you can write "Based on the data and statistical questions, the following figures are recommended:"

File: report/main.qmd [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+
+1.  Side-by-side boxplots of zero-calorie or sugary drink sales by variables such as day of the week or site can help identify if there are differences in sales between different levels of the categorical variables.
+
+2.  Plots of the proportion of sugary and zero-calorie sales coming from zero-calore drinks over counts, such as Figure 1, can be used to observe the pattern over time. These plots can also be stratified by other variables, such as intervention to identify differences between groups.
```
Comment (sarahmasri) [inline report/main.qmd:53]: zero-calorie instead of zero-calore

File: report/main.qmd [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+
+3.  A scatterplot of zero-calorie vs sugary drink sales be used to investigate what relationship they have and potentially identify if they are highly correlated.
+
+Additionally, summary statistics such as means can be calculated for different variables, such as average zero-calorie sales for each day of the week to give a numeric summary.
+
```
Comment (sarahmasri) [inline report/main.qmd:58]: Correlation plot might be useful as well. Also, a plot to visualize missing data/balance between factors

File: report/main.qmd [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+ggplot(aes(x=Count,y=percent_zero,col=Site))+
+geom_point()+
+geom_smooth(se=F)+
+scale_fill_continuous(guide = guide_legend()) +
+theme(legend.position="bottom")+
+labs(x = "Day of the Experiment",
+     y = "Proportion of Zero-Calorie \nand Sugary Drink Sales")
+```
+
+# Formal Analysis
```
Comment (sarahmasri) [inline report/main.qmd:80]: How do you recommend to handle the missing data?

File: report/main.qmd [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+
+In this study, an effective intervention will increase the proportion of daily sales that come from zero-calorie beverages among sales that are either zero-calorie beverages or sugary beverages. This proportion increases when there is relatively more sales of zero-calorie beverages than of sugary beverages. This avoids the need to perform two separate analyses (one for zero-calorie and another one for sugary beverages) and any variation due to sales of other unrelated products.
```
Comment (sarahmasri) [inline report/main.qmd:82]: (First sentence) Increase relative to the baseline data?

File: report/main.qmd [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+
+To answer the statistical questions, we recommend using a generalized linear mixed effects model (GLMM) with a Poisson distribution. This type of model uses count data and allows for different effects across sites. Another advantage of this model is that it can handle the longitudinal nature of the data. The response variable is the counts of zero-calorie beverage sales. Including the log-transformed total sales of zero-calorie or sugary beverages as an offset, this models the proportion of zero-calorie beverage sales as defined previously. The main independent variables of interest are intervention indicators, treating the baseline periods as the reference. Other covariates in the model include the site, the day of the week, and the study day.
```
Comment (sarahmasri) [inline report/main.qmd:84]: GEE might also be good to consider here. It can be thought of as an extension of GLMM to longitudinal data since it can handle some missing data and has no distributional assumptions. Also it can address serial correlation.

File: report/main.qmd [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+
+To assess whether intervention effects differ by site, we recommend testing interaction effects between sites and intervention groups. This assessment can be implemented as a Likelihood Ratio Test. We recommend including random effects for intervention groups, which will allow for different effects between groups.
```
Comment (sarahmasri) [inline report/main.qmd:86]: This is good

File: report/main.qmd [hunk 14]
```diff
@@ -0,0 +1,103 @@
...
+
+A major assumption of the GLMM with Poisson model is the correct specification of the Poisson distribution. To avoid this assumption, a robust version of the above model is recommended. In particular, the same model specification can be used, while employing Poisson estimating equations with a so-called sandwich covariance estimator. This model is known as robust Poisson as it remains valid even if the data do not follow a Poisson distribution @Talbot2022. Once exponentiated, coefficients from the GLMM and the robust Poisson model have the same interpretation. For instance, if the exponentiated coefficient for a given intervention is equal to 1.5, then the proportion of zero-calorie beverage sales increases by 50% under the intervention, as compared to the baseline.
+
```
Comment (sarahmasri) [inline report/main.qmd:89]: It might be useful to also recommend a secondary method to compare results. LME methods are also applicable here.

**Review by sarahmasri — COMMENTED** [review]
Great report!

File: report/main.qmd [hunk 15]
```diff
@@ -0,0 +1,103 @@
...
+sugary_bev %>%
+mutate(percent_zero = ZeroCal/(ZeroCal+Sugary)) %>%
+ggplot(aes(x=Count,y=percent_zero,col=Site))+
+geom_point()+
+geom_smooth(se=F)+
+scale_fill_continuous(guide = guide_legend()) +
+theme(legend.position="bottom")+
+labs(x = "Day of the Experiment",
+     y = "Proportion of Zero-Calorie \nand Sugary Drink Sales")
+```
```
Comment (JohnsonChen-id) [inline report/main.qmd:78]: A description of relationships observed in the scatterplot might be useful here. 
Plot looks great, maybe decrease alpha by a bit so the points look less packed

File: report/main.qmd (hunk 12 above)
Comment (JohnsonChen-id) [inline report/main.qmd:84]: A statement about how GLMM handles missing data within the model would be helpful for the client. 
The client sounds familiar with or at least knows about LMM, so it would be helpful to introduce GLMM at least building from the understanding of LMM. 
You may also want to explain how to incorporate the proportion of zero-calorie drink sales into a GLMM poison model to the client.

File: report/main.qmd [hunk 16]
```diff
@@ -0,0 +1,103 @@
...
+
+To assess whether intervention effects differ by site, we recommend testing interaction effects between sites and intervention groups. This assessment can be implemented as a Likelihood Ratio Test. We recommend including random effects for intervention groups, which will allow for different effects between groups.
+
+A major assumption of the GLMM with Poisson model is the correct specification of the Poisson distribution. To avoid this assumption, a robust version of the above model is recommended. In particular, the same model specification can be used, while employing Poisson estimating equations with a so-called sandwich covariance estimator. This model is known as robust Poisson as it remains valid even if the data do not follow a Poisson distribution @Talbot2022. Once exponentiated, coefficients from the GLMM and the robust Poisson model have the same interpretation. For instance, if the exponentiated coefficient for a given intervention is equal to 1.5, then the proportion of zero-calorie beverage sales increases by 50% under the intervention, as compared to the baseline.
```
Comment (JohnsonChen-id) [inline report/main.qmd:88]: Great paragraph

**Review by JohnsonChen-id — COMMENTED** [review]
Great Report!

File: report/main.qmd (hunk 7 above)
Comment (parpishro) [inline report/main.qmd:49]: Note also that we are not recommending how client can describe data. It is actually doing the data description/summary.

File: report/main.qmd [hunk 17]
```diff
@@ -0,0 +1,103 @@
...
+library(here)
+library(tidyverse)
+# Read in data
+sugary_bev <- read.csv(file = here("./rawdata/june1data.csv"),
+colClasses = c("DofW" = "factor", "Site" = "factor", "Intervention" = "factor"))
+# Plot percentage of sugary or zero sugar sales accounted for by zero sugar drinks over the counts by site
+sugary_bev %>%
+mutate(percent_zero = ZeroCal/(ZeroCal+Sugary)) %>%
+ggplot(aes(x=Count,y=percent_zero,col=Site))+
+geom_point()+
```
Comment (parpishro) [inline report/main.qmd:72]: Recommend using a colour-blind friendly colour  palette. This can be easily done by using:

- installing and loading the viridis package
- adding to `scale_fill_viridis(discrete = TRUE)` to your ggplot sentence.

File: report/main.qmd (hunk 3 above)
Comment (parpishro) [inline report/main.qmd:41]: stating sample size (631) is important.

File: report/main.qmd [hunk 18]
```diff
@@ -0,0 +1,103 @@
...
+scale_fill_continuous(guide = guide_legend()) +
+theme(legend.position="bottom")+
+labs(x = "Day of the Experiment",
+     y = "Proportion of Zero-Calorie \nand Sugary Drink Sales")
+```
+
+# Formal Analysis
+
+In this study, an effective intervention will increase the proportion of daily sales that come from zero-calorie beverages among sales that are either zero-calorie beverages or sugary beverages. This proportion increases when there is relatively more sales of zero-calorie beverages than of sugary beverages. This avoids the need to perform two separate analyses (one for zero-calorie and another one for sugary beverages) and any variation due to sales of other unrelated products.
+
```
Comment (parpishro) [inline report/main.qmd:83]: A suitable model for this study can be Interrupted Time Series Analysis, which is specifically designed to handle the  interrupted time-series multi-site quasi-experimental study designs. Based on my research, these are the few reasons why ITSA should be considered primary model for this type of study:

Interrupted Time Series Analysis (ITSA) is apt for interrupted time-series multi-site quasi-experimental designs due to:

1. Quasi-experimental Design: Directly models pre- and post-intervention effects, enabling causal inference without random assignment.
2. Intervention Impact Analysis: Analyzes immediate and long-term intervention impacts, crucial for multi-site variations.
3. Adjustments for Trends and Seasonality: Accounts for underlying trends and seasonal patterns, preventing misattribution of intervention effects.
4. Autocorrelation Management: Incorporates methods like ARIMA to handle autocorrelation, ensuring accurate effect estimation.
5. Site Comparability: Facilitates analysis across multiple sites, allowing for aggregation or comparison of intervention effects.
6. Policy Evaluation: Well-suited for evaluating policy or program effectiveness over time.

Most of above points are applicable to this study.

File: report/main.qmd (hunk 10 above)
Comment (parpishro) [inline report/main.qmd:80]: I think the formal analysis, should follow naturally from Data EDA sections: data properties and study design narrows down potential applicable models and visualizations, summaries, univariate analyses in EDA guide hypothesis formulation and/or variable selection.  tThis link between EDA and Data on one hand and Formal analysis on the other hand can be made stronger. One way to do that is to perform some univariate analyses and visualize in different forms.

File: report/main.qmd [hunk 19]
```diff
@@ -0,0 +1,103 @@
...
+Therefore, we recommend fitting a Poisson GLMM to the log-transformed sales to predict the proportion of zero-calorie beverage sales, and determining which effects are most relevant using the likelihood ratio. However, the Poisson GLMM assumes that the underlying data follows a Poisson distribution. If this assumption is invalid, we recommend utilizing robust Poisson regression, as the robust Poisson model does not require this distributional assumption. Both analyses employ the zero-calorie beverage sales as the primary outcome with the log of the sum of zero-calorie beverage sales and sugary beverage sales as an offset.
+
+{{< pagebreak >}}
+
+# References
+
+::: {#refs}
+:::
+
+# Statistical Appendix
```
Comment (parpishro) [inline report/main.qmd:103]: If visualization and tables takes up too much space, you can include them here and reference them in text if needed. Moreover, you can include more in-depth model introduction here.

File: code/01-exploration.R [hunk 20]
```diff
@@ -0,0 +1,96 @@
+library(here)
+library(tidyverse)
+
```
Comment (parpishro) [inline code/01-exploration.R:3]: a box-plot of different interventions can be also informative.

File: code/01-exploration.R [hunk 21]
```diff
@@ -0,0 +1,96 @@
...
+# Read in data
+sugary_bev <- read.csv(file = "./rawdata/june1data.csv", 
+  colClasses = c("DofW" = "factor", "Site" = "factor", "Intervention" = "factor"))
+
```
Comment (parpishro) [inline code/01-exploration.R:7]: there is no exploration of missing data. As a starter, you can use visdat package to visualize the missingness.

**Review by parpishro — APPROVED** [review]

File: report/main.qmd [hunk 22]
```diff
@@ -0,0 +1,103 @@
...
+```
+
+# Introduction
+
+Strong evidence has linked habitual intake of sugar-sweetened beverages (SSBs) with weight gain and a higher risk of type 2 diabetes mellitus, cardiovascular diseases, and even cancer @Malik2022. Multiple policy interventions to inhibit consumption of SSBs have been proposed @Krieger2021, including taxation of SSBs, limiting access to SSBs in schools and healthcare facilities, and adding informative package labelling. As policymakers create potential interventions to limit SSB consumption, there is a need for data-driven methods of determining their efficacy.
+
+This study aims to evaluate five interventions which intend to shift consumer behaviour towards zero-calorie beverages. These interventions include three information-based interventions and two financial interventions.
+
+The main statistical question is whether these interventions are associated with differences in the average proportion of daily sales of zero-calorie beverages. Secondary questions include whether there are significant interactions between interventions and site effects, and whether a combination of interventions was more effective than the individual interventions.
```
Comment (dajmcdon) [inline report/main.qmd:37]: Is "proportion" really the right target?

File: report/main.qmd [hunk 23]
```diff
@@ -0,0 +1,103 @@
...
+# Plot percentage of sugary or zero sugar sales accounted for by zero sugar drinks over the counts by site
+sugary_bev %>%
+mutate(percent_zero = ZeroCal/(ZeroCal+Sugary)) %>%
+ggplot(aes(x=Count,y=percent_zero,col=Site))+
+geom_point()+
+geom_smooth(se=F)+
+scale_fill_continuous(guide = guide_legend()) +
+theme(legend.position="bottom")+
+labs(x = "Day of the Experiment",
+     y = "Proportion of Zero-Calorie \nand Sugary Drink Sales")
```
Comment (dajmcdon) [inline report/main.qmd:77]: I don't understand this axis label. How can it be a proportion of both simultaneously?

**Review by dajmcdon — APPROVED** [review]
* I went fast, so some comments may sound harsh, but they aren't meant to be.
  Just to draw your attention.
* The general model (Poisson with offset) is good. But I don't understand "Robust Poisson". 
    * Your description makes me think it's nonsense. In statistics, we can say that a model "is robust to misspecification" meaning "it may fit well even if some of the assumptions are not quite satisfied". 
    * Or "robust to outliers" meaning "if there are a few observations with large residuals, the parameter estimates won't be affected". This typically requires a "robust loss" like Huber or Absolute Error. 
    * Finally, you could use the "`quasipoisson`" link which allows for overdispersion (allows the mean=sd relation of Poisson to be relaxed).
    * Sandwich covariance is something else entirely.
* So, either this explanation is wrong, or it's not clear to me (and definitely not to the client).
